channel/src/test/java/com/synopsys/integration/alert/channel/slack/distribution/SlackDistributionEventReceiverTest.java

```diff
@@ -0,0 +1,127 @@
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.alert.common.persistence.accessor.SlackJobDetailsAccessor;
+import com.synopsys.integration.alert.common.persistence.model.job.details.SlackJobDetailsModel;
+import com.synopsys.integration.alert.common.rest.ProxyManager;
+import com.synopsys.integration.alert.common.util.MarkupEncoderUtil;
+import com.synopsys.integration.alert.descriptor.api.SlackChannelKey;
+import com.synopsys.integration.alert.descriptor.api.model.ChannelKeys;
+import com.synopsys.integration.alert.processor.api.distribute.DistributionEvent;
+import com.synopsys.integration.alert.processor.api.extract.model.ProviderDetails;
+import com.synopsys.integration.alert.processor.api.extract.model.ProviderMessageHolder;
+import com.synopsys.integration.alert.processor.api.extract.model.SimpleMessage;
+import com.synopsys.integration.alert.test.common.MockAlertProperties;
+import com.synopsys.integration.rest.proxy.ProxyInfo;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+public class SlackDistributionEventReceiverTest {
+    private static final Set<Long> FIRST_MESSAGE_NOTIFICATION_IDS = Set.of(1L, 2L, 3L);
+    private static final Set<Long> SECOND_MESSAGE_NOTIFICATION_IDS = Set.of(4L, 5L, 6L);
+
+    private final SlackChannelKey slackChannelKey = new SlackChannelKey();
+
+    private SlackDistributionEventReceiver slackDistributionEventReceiver;
+    private MockProcessingAuditAccessor processingAuditAccessor = new MockProcessingAuditAccessor();
+    private MockWebServer mockSlackServer = new MockWebServer();
+
+    @BeforeEach
+    public void init() throws IOException {
+        Gson gson = new Gson();
+
+        MarkupEncoderUtil markupEncoderUtil = new MarkupEncoderUtil();
+        SlackChannelMessageFormatter slackChannelMessageFormatter = new SlackChannelMessageFormatter(markupEncoderUtil);
+        SlackChannelMessageConverter slackChannelMessageConverter = new SlackChannelMessageConverter(slackChannelMessageFormatter);
+        SlackChannelMessageSender slackChannelMessageSender = new SlackChannelMessageSender(createRestChannelUtility(), ChannelKeys.SLACK);
+        SlackChannel slackChannel = new SlackChannel(slackChannelMessageConverter, slackChannelMessageSender);
+
+        mockSlackServer.start();
+        String url = mockSlackServer.url("/").toString();
+
+        SlackJobDetailsModel slackJobDetailsModel = new SlackJobDetailsModel(null, url, "channelName", "userName");
+
+        SlackJobDetailsAccessor slackJobDetailsAccessor = jobId -> Optional.of(slackJobDetailsModel);
+
+        slackDistributionEventReceiver = new SlackDistributionEventReceiver(gson, processingAuditAccessor, slackJobDetailsAccessor, slackChannel, slackChannelKey);
+    }
+
+    @AfterEach
+    public void cleanup() throws IOException {
+        mockSlackServer.shutdown();
+    }
```

Comment:
Should we construct/destruct this once rather than once per test?

Reply:
We created it this way to ensure that the MockWebServer queue is empty between tests so that if the first test was to fail for some reason it wouldn't impact the second test with responses still in queue.

Reply:
There's no way to clear the queue instead?

Reply:
I don't see a way immediately to remove it using MockWebServer's api. Even if we did though I believe this would be an issue for our mockSlackServer.getRequestCount assertions that we are doing in both tests